test/jdk/java/foreign/CallGeneratorHelper.java

```diff
@@ -378,11 +378,11 @@ static void generateUpcallFunction(String fName, Ret ret, List<ParamType> params
     @SuppressWarnings("unchecked")
     static Object makeArg(MemoryLayout layout, List<Consumer<Object>> checks, boolean check) throws ReflectiveOperationException {
         if (layout instanceof GroupLayout) {
-            MemorySegment segment = MemorySegment.allocateNative(layout, MemorySession.openImplicit());
+            MemorySegment segment = MemorySession.openImplicit().allocate(layout);
```

Comment:
This (and other similar calls) can be simplified to:

`MemorySegment.allocateNative(layout)`

I suggest looking for the pattern `MemorySession.openImplicit().allocate`